library/axi_ad7606x/axi_ad7606x_ip.tcl

```diff
@@ -46,60 +46,5 @@ set cc [ipx::current_core]
 set_property display_name "AXI AD7606X" $cc
 set_property description "AXI AD7606X HDL interface" $cc
 
-## define external_clk port as clock interface
-adi_add_bus external_clk slave \
-  "xilinx.com:signal:clock_rtl:1.0" \
-  "xilinx.com:signal:clock:1.0" \
-  [list {"external_clk" "CLK"} ]
-
-adi_set_ports_dependency "external_clk" \
-  "(spirit:decode(id('MODELPARAM_VALUE.EXTERNAL_CLK')) = 1)" 0
-
-## parameter validation
-
-set_property -dict [list \
-  "value_format" "bool" \
-  "value" "true" \
-  ] \
-[ipx::get_hdl_parameters EXTERNAL_CLK -of_objects $cc]
-
-## customize XGUI layout
-
-## remove the automatically generated GUI page
-
-ipgui::remove_page -component $cc [ipgui::get_pagespec -name "Page 0" -component $cc]
-ipx::save_core $cc
-
-## create a new GUI page
-
-ipgui::add_page -name {AXI AD7606X} -component $cc -display_name {AXI AD7606X}
-set page0 [ipgui::get_pagespec -name "AXI AD7606X" -component $cc]
-
-ipgui::add_param -name "EXTERNAL_CLK" -component $cc -parent $page0
-set_property -dict [list \
-  "display_name" "EXTERNAL_CLK" \
-  "tooltip" "External clock for the ADC" \
-  "widget" "checkBox" \
-] [ipgui::get_guiparamspec -name "EXTERNAL_CLK" -component $cc]
-
-ipgui::add_param -name "DEV_CONFIG" -component $cc -parent $page0
```

Comment:
To offer an UI-based alternative for the setup of the AXI-based AD7606X IP, there would be better to keep a page that exposes the **ADC_N_BITS** parameter.

Reply:
In this PR, I have just cherry-picked some commits from 'main' to 'next stable'. 
Anny extra fix / change should be pushed first to 'main' and then I will cherry-pick it to 'next_stable'.

Reply:
@SRaus ok